components/Nav/NavMobile/NavMobile.js

```diff
@@ -0,0 +1,87 @@
+import React, { Component } from 'react';
+import Link from 'next/link';
+import { donateLink, s3 } from 'common/constants/urls';
+import { navItems } from 'common/constants/navigation';
+import flattenDepth from 'lodash/flattenDepth';
+import HamburgerIcon from 'static/images/icons/hamburger.svg';
+import styles from './NavMobile.css';
+
+export default class NavMobile extends Component {
+  constructor(props) {
+    super(props);
+    this.handleOnClick = this.handleOnClick.bind(this);
+    this.handleMouseLeave = this.handleMouseLeave.bind(this);
+    this.handleKeyPress = this.handleKeyPress.bind(this);
+  }
+
+  state = {
+    dropdownShow: false,
+  };
+
+  // flattens navItems
+  links = () => flattenDepth(navItems.map(navItem => [navItem, navItem.sublinks]), 2);
+
+  toggleDropdown() {
+    this.setState(prevState => ({ dropdownShow: !prevState.dropdownShow }));
+  }
+
+  handleOnClick() {
+    this.toggleDropdown();
+  }
+
+  handleMouseLeave() {
+    this.toggleDropdown();
+  }
+
+  handleKeyPress(e) {
+    if (e.key === 'Enter') {
+      this.toggleDropdown();
+    }
+  }
+
+  render() {
+    const { state, handleOnClick, handleMouseLeave, handleKeyPress, links } = this;
```

Comment:
I don't mind destructuring state or props from `this`, but I'd prefer to reference the methods straight off of this. It at least makes it clear they're methods.